lambda-buffers-compiler/test/Test/KindCheck.hs

```diff
@@ -46,14 +60,35 @@ kcTestFailing =
     assertBool "Test should have failed." $
       check_ compilerInput'incoherent /= Right ()
 
+{- | TyDef order does not matter when kind checking.
+
+ We're not interested in the failure error as there might be more than two
+ errors in a module - and it is non-determistic which one is first. But it is
```

Comment:
You could just use Set on errors and the `Eq` on them ignores the order.

Reply:
Not really - the compiler throws one error and it stops. That would mean that the compiler would have to find all errors - which is not what it is doing atm.